### PR TITLE
feat: implement OpenTelemetry tracing

### PR DESCRIPTION
**Pull request by SergeTupchiy**
Closes EMQX-10535

### <samp>🤖 Generated by Copilot at e36685c</samp>

This pull request adds Opentelemetry tracing to the MQTT message lifecycle in the `emqx` application. It modifies the `emqx_broker`, `emqx_channel`, and `emqx_connection` modules to create and propagate trace spans and events for each message processing and delivery step. It also adds a new module `emqx_otel_trace` to implement the tracing logic and a new field `extra` to the `message` and `mqtt_packet` records to store the trace context. The purpose of this change is to enable observability and performance analysis of the MQTT broker using Opentelemetry.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

**Comment by SergeTupchiy** [comment]
Quite rough draft to accomplish RFC I'm currently working on it.

File: apps/emqx/src/emqx_broker.erl [hunk 1]
```diff
@@ -555,8 +567,10 @@ do_dispatch(Topic, #delivery{message = Msg}) ->
         0 ->
             ok = emqx_hooks:run('message.dropped', [Msg, #{node => node()}, no_subscribers]),
             ok = inc_dropped_cnt(Msg),
+            emqx_otel_trace:event(message_dropped),
```
Comment (savonarola) [inline apps/emqx/src/emqx_broker.erl:570]: Would it make sense to integrate otel events into emqx metrics? May be specifying the subset of metrics denoted for otel export as events.

Comment (SergeTupchiy) [reply apps/emqx/src/emqx_broker.erl:570]: Thanks for the suggestion.
I added this event more like an example.
Some events may be not related to any of the emqx metrics, some may be not worth measuring.
I assume we can consider integrating events to metrics once we have more or less stable list of events we want to report..

File: apps/emqx/etc/acl.conf
Comment (thalesmg) [inline apps/emqx/etc/acl.conf:1]: Q: was this file added by accident?  It seems unrelated to the PR.  :thinking:

Comment (SergeTupchiy) [reply apps/emqx/etc/acl.conf:1]: Yes, added it accidentally. :see_no_evil:, 
Thanks, removed.

**Comment by SergeTupchiy** [comment]
Closing it, as it's quite stale.. I'm going to open a new PR soon.

**Pull request closed without merging**